DigitalLearningSolutions.Web/ViewModels/MyAccount/EditDetailsViewModel.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.MyAccount
+{
+    using System.ComponentModel.DataAnnotations;
+    using DigitalLearningSolutions.Data.Models.User;
+
+    public class EditDetailsViewModel
+    {
+        public EditDetailsViewModel() { }
+
+        public EditDetailsViewModel(
+            AdminUser? adminUser,
+            DelegateUser? delegateUser)
+        {
+            FirstName = adminUser?.FirstName ?? delegateUser?.FirstName;
+            LastName = adminUser?.LastName ?? delegateUser?.LastName;
+            Email = adminUser?.EmailAddress ?? delegateUser?.EmailAddress;
+        }
+
+        [Required(ErrorMessage = "Enter a first name.")]
+        public string? FirstName { get; set; }
+
+        [Required(ErrorMessage = "Enter a last name.")]
+        public string? LastName { get; set; }
+
+        [Required(ErrorMessage = "Enter an email address.")]
+        [EmailAddress(ErrorMessage = "Enter a valid email address.")]
+        public string? Email { get; set; }
+
+        [Required(ErrorMessage = "Enter your password.")]
+        [DataType(DataType.Password)]
+        public string? Password { get; set; }
```

Comment:
Could we add a maxLength of 100 constraint to all of these fields? 🙏  

[MaxLength(100, ErrorMessage = "First name must be at most 100 characters")]
[MaxLength(100, ErrorMessage = "Last name must be at most 100 characters")]
[MaxLength(100, ErrorMessage = "Email address must be at most 100 characters")]

Less concerned about PW as we're checking against existing hash, but feel free to add that:
[MaxLength(100, ErrorMessage = "Password must be at most 100 characters")]